sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -2287,6 +2336,16 @@ pi_result piContextRelease(pi_context Context) {
   return PI_SUCCESS;
 }
 
+pi_result piContextRelease(pi_context Context) {
+  pi_platform Plt = Context->Devices[0]->Platform;
+  std::unique_lock<std::mutex> ContextsLock(Plt->ContextsMutex,
```

Comment:
This lock is redundant if we aren't doing the tracking, right? Can this be avoided?

Reply:
I see avoid the redundancy I mentioned by only locking in the if just below. But then it's too much inter-knowledge between the functions.

Reply:
If you can't have the "helper" to always lock by itself, maybe pass an argument if the lock is already done or not?

Reply:
If we aren't doing the tracking then it is not locked because std::defer_lock is used. std::unique_lock object is used just to define the scope of the lock, otherwise we will need to work with mutex itself and call lock/unlock manualy. Destructor of std::unique_lock will unlock  the mutex only if it was locked.

Reply:
> maybe pass an argument if the lock is already done or not?

How about this?